tests/FSharp.Compiler.ComponentTests/Conformance/UnionTypes/MultiCaseUnionStructTypes.fs

```diff
@@ -0,0 +1,631 @@
+            (Error 3204, Line 5, Col 21, Line 5, Col 23, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+        ]
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 40`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of a: int * a1: string
+    | B of b: string
+    | C of string
+        """
+        |> compile
+        |> shouldSucceed
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 41`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of int * string
+    | B of string
+    | C of c: string
+        """
+        |> compile
+        |> shouldSucceed
+
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 42`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of int * string
+    | B of b: string
+    | C of string
+        """
+        |> compile
+        |> shouldSucceed
+        
+    [<Fact>]
+    let ``If a union type has more than one case and is a struct, field must be given unique name 43`` () =
+        Fsx """
+namespace Foo
+[<Struct>]
+type StructUnion =
+    | A of a: int * string
+    | B of string * b: string
+    | C of c: string * string * c3: int
+        """
+        |> compile
+        |> shouldFail
+        |> withDiagnostics [
+            (Error 3204, Line 5, Col 12, Line 5, Col 13, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 5, Col 21, Line 5, Col 27, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 6, Col 12, Line 6, Col 18, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 6, Col 21, Line 6, Col 22, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 7, Col 12, Line 7, Col 13, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+            (Error 3204, Line 7, Col 33, Line 7, Col 35, "If a multicase union type is a struct, then all union cases must have unique names. For example: 'type A = B of b: int | C of c: int'.")
+        ]
+        
+    [<Fact>]
```

Comment:
This is such an exemplary testing that I will even happily forgive you the naming :D

Reply:
Naming is hard you know  😅.